### PR TITLE
LibWeb: Add support for animation-timing-function

### PR DESCRIPTION
**Pull request by alimpfard**
_smooooooooooooooooooooooooooth_ (when blitting images isn't eating CPU cycles)

### CODE REVIEW

File: Meta/Lagom/Tools/CodeGenerators/LibWeb/GenerateCSSEasingFunctions.cpp [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+    TRY(generate_header_file(easing_data, *generated_header_file));
+    TRY(generate_implementation_file(easing_data, *generated_implementation_file));
+
+    return 0;
+}
+
+static ErrorOr<String> title_casify_easing_function(StringView input)
+{
+    // Easing functions are named in kebab-case, turn them into TitleCase.
```
Comment (AtkinsSJ) [inline Meta/Lagom/Tools/CodeGenerators/LibWeb/GenerateCSSEasingFunctions.cpp:44]: This is what the existing `title_casify()` function does. All the ones in GeneratorUtil work on kebab-case inputs. (Probably they should have better names.)

Comment (alimpfard) [reply Meta/Lagom/Tools/CodeGenerators/LibWeb/GenerateCSSEasingFunctions.cpp:44]: Oh I see, will fix.

**Review by AtkinsSJ — CHANGES_REQUESTED** [review]
Yay animations! :tada:

Side note: I'm going to have to pull the "parse `<foo [a,b]>?`" code out into a helper soon, we're doing that in several generators now.

File: Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp [hunk 2]
```diff
@@ -7079,6 +7080,116 @@ ErrorOr<RefPtr<StyleValue>> Parser::parse_text_decoration_line_value(TokenStream
...
+            if (argument_tokens.has_next_token()) {
+                // Arguments must be separated by commas.
+                if (!argument_tokens.next_token().is(Token::Type::Comma))
+                    return nullptr;
+                argument_tokens.skip_whitespace();
+
+                // If there are no more parameters after the comma, this is invalid.
+                if (!argument_tokens.has_next_token())
+                    return nullptr;
+            }
```
[the remark this answers is not in the document]
Comment (alimpfard) [reply Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp:7178]: :thinking: I copied that over from TransformFunction's parsing code, should I rewrite that as well?

File: Meta/Lagom/Tools/CodeGenerators/LibWeb/GenerateCSSPropertyID.cpp [hunk 3]
```diff
@@ -640,6 +644,8 @@ bool property_accepts_type(PropertyID property_id, ValueType value_type)
                     TRY(property_generator.try_appendln("        case ValueType::Time:"));
                 } else if (type_name == "url") {
                     TRY(property_generator.try_appendln("        case ValueType::Url:"));
+                } else if (type_name == "easing-function") {
```
[the remark this answers is not in the document]
Comment (alimpfard) [reply Meta/Lagom/Tools/CodeGenerators/LibWeb/GenerateCSSPropertyID.cpp:647]: OK

File: Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (AtkinsSJ) [reply Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp:7178]: Correction: Git very decidedly points fingers at me. :laughing:

[the remark this answers is not in the document]
Comment (alimpfard) [reply Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp:7178]: I'll rewrite that code, and push it here if it doesn't blow up :P

**Pull request merged**